cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/cost/AzurePricingCache.java

```diff
@@ -0,0 +1,118 @@
+package com.sequenceiq.cloudbreak.cloud.azure.cost;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.sequenceiq.cloudbreak.cloud.azure.cost.model.PriceResponse;
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmTypes;
+import com.sequenceiq.cloudbreak.cloud.model.ExtendedCloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.VmType;
+import com.sequenceiq.cloudbreak.cloud.model.VmTypeMeta;
+import com.sequenceiq.cloudbreak.cloud.service.CloudParameterService;
+import com.sequenceiq.common.api.type.CdpResourceType;
+
+@Service("azurePricingCache")
+public class AzurePricingCache {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzurePricingCache.class);
+
+    private static final String PRICING_API_REQUEST_TEMPLATE = "https://prices.azure.com/api/retail/prices?$filter=%s";
+
+    private static final int MAX_ATTEMPTS = 5;
+
+    private static final int BACKOFF_DELAY = 500;
+
+    private final Table<String, String, PriceResponse> priceCache = HashBasedTable.create();
+
+    private final Table<String, String, VmTypeMeta> vmCache = HashBasedTable.create();
+
+    @Inject
+    private CloudParameterService cloudParameterService;
+
+    public double getPriceForInstanceType(String region, String instanceType) {
```

Comment:
we also some kind of caching here

Reply:
In the `AzurePricingCache` there is a separate cache for pricing information and vm metadata information.
The `getPriceResponse` and `getVmMetadata` methods interact with these caches.